conformance/Makefile

```diff
@@ -36,16 +44,20 @@ install-nkg: ## Install NKG with provisioner on configured kind cluster
 	kubectl apply -f ../deploy/manifests/rbac.yaml
 	kubectl apply -f ../deploy/manifests/gatewayclass.yaml
 	kubectl apply -f ../deploy/manifests/service/nodeport.yaml
-	kubectl apply -f provisioner/provisioner.yaml
-
-.PHONY: update-test-kind-config 
-update-test-kind-config: ## Update kind config
-	sed -ir "s|server:.*|server: https://kind-control-plane:6443|" $(KIND_KUBE_CONFIG_FOLDER)/config
+	yq '(select(di != 3))' provisioner/provisioner.yaml | kubectl apply -f -
+	yq '(select(.spec.template.spec.containers[].image) | .spec.template.spec.containers[].image="$(NKG_PREFIX):$(NKG_TAG)" | .spec.template.spec.containers[].imagePullPolicy = "Never")' provisioner/provisioner.yaml | kubectl apply -f -
 
 .PHONY: run-conformance-tests
-run-conformance-tests: update-test-kind-config ## Run conformance tests
-	docker run --network=kind --rm -v $(KIND_KUBE_CONFIG_FOLDER):/root/.kube $(PREFIX):$(TAG) \
-		go test -timeout 25m -v . -tags conformance -args --gateway-class=$(GATEWAY_CLASS) --debug --supported-features=$(SUPPORTED_FEATURES) --exempt-features=$(EXEMPT_FEATURES)
+run-conformance-tests: ## Run conformance tests
```

Comment:
I wonder if it's gonna be more convinient to run the pod using `kubectl run` command. For example:
```
kubectl run -i --tty conformance --image=conformance-test-runner:latest --image-pull-policy=Never --restart=Never --overrides='{ "spec": { "serviceAccount": "conformance" }  }' -- go test -timeout 25m -v . -tags conformance -args --gateway-class=nginx --debug --supported-features=HTTPRoute,HTTPRouteQueryParamMatching,HTTPRouteMethodMatching,HTTPRoutePortRedirect,HTTPRouteSchemeRedirect --exempt-features=ReferenceGrant
```

This way we don't need to rebuild the container if we need to change anything in how we run the tests. Additionally, we can keep the existing GATEWAY_CLASS, SUPPORTED_FEATURES, EXEMPT_FEATURES variables of the Makefile for quick customization.